### DISCUSSION TITLE
completition sources / completition priority

### DISCUSSION THREAD

**Discussion by Shuraken007** [neoclide/coc.nvim#4799] [Q&A] [answered]
Hello. I'm configuring coc for c#.

Lot of results messed up completion menu. They gives me results far away from that I want.

Is it possible to have at first - default values, without lsp-server:
![image](https://github.com/neoclide/coc.nvim/assets/23379636/3eddbadf-e924-41b8-a4ed-a1ef27203562)

And only after show smth from server without duplicating:
![image](https://github.com/neoclide/coc.nvim/assets/23379636/14c365c7-bbd5-42ef-99f5-dc22a1d99b38)

I have `csharp-ls` at coc-settings:
```json
  "languageserver": {
    "csharp-ls": {
      "command": "csharp-ls",
      "filetypes": [
        "cs"
      ],
      "rootPatterns": [
        "*.csproj",
        ".vim/",
        ".git/",
        ".hg/"
      ]
    }
}
```

In other words - I want completion from different sources at the same time: ['default', 'csharp-ls'], not sorted, or sorted source by source, not mixed. Ideally to have smth like ['language keywords', 'project tags (existing function/classes/variables)', 'rest from lsp']

**Comment by Shuraken007** [comment, marked as answer]
Well, I founded sources / priority options.

Now it looks much more comfortable.
![image](https://github.com/neoclide/coc.nvim/assets/23379636/bc33e697-9fe5-4035-8403-3cd29ef9969d)
![image](https://github.com/neoclide/coc.nvim/assets/23379636/a6e15e3c-263b-4605-af15-1587356660ff)

with extension coc-snippets, coc-tag

```json
  "snippets.priority": 400,
  "coc": {
    "source": {
      "around": {
        "enable": 1,
        "priority": 400
      },
      "buffer": {
        "enable": 1,
        "priority": 400
      },
      "file": {
        "enable": 1,
        "priority": 300
      },
      "tag": {
        "enable": 1,
        "priority": 300
      }
    }
  },
```
